prescient/data/providers/shortcut_data_provider.py

```diff
@@ -0,0 +1,319 @@
+
+        Notes
+        -----
+        This will store actuals data in the model's existing data arrays, starting
+        at index 0.  If the model's arrays are not big enough to hold all the
+        requested time steps, only those steps for which there is sufficient storage
+        will be saved.  If arrays are larger than the number of requested time 
+        steps, the remaining array elements will be left unchanged.
+
+        Note that this method has the same signature as populate_with_forecast_data.
+        '''
+        self._populate_with_forecastable_data('real_time', start_time, num_time_periods, time_period_length_minutes, model)
+
+    def _populate_with_forecastable_data(self,
+                                         sim_type:str,
+                                         start_time:datetime,
+                                         num_time_periods: int,
+                                         time_period_length_minutes: int,
+                                         model: EgretModel
+                                        ) -> None:
+                                                                                     # pandas is inclusive on ranges
+        end_time = start_time + timedelta(minutes=num_time_periods*time_period_length_minutes) - timedelta(seconds=1)
+
+        price_data = self._historical_prices[sim_type][start_time:end_time].to_list()
+
+        if sim_type == 'day_ahead':
+            step_ratio = 1
+        else:
+            step_ratio = time_period_length_minutes // self._frequency_minutes
+
+        if step_ratio == 1 and len(model.data['system']['time_keys']) == num_time_periods:
+            p_cost = model.data['elements']['generator']['system']['p_cost']['values'] 
+            for i, val in enumerate(price_data):
+                p_cost[i] = val
+        else:
+            copy_from = model.clone()
+            copy_from.data['elements']['generator']['system']['p_cost']['values'] = price_data
+            # if the data is at a higher frequency, it is more sensible
+            # to average over the time periods for prices
+            _recurse_average_at_ratio(copy_from.data, model.data, step_ratio)
+
+        # Fill in times:
+        time_labels = model.data['system']['time_keys']
+        delta = timedelta(minutes=time_period_length_minutes)
+        for i in range(len(time_labels)):
+            dt = start_time + i*delta
+            time_labels[i] = dt.strftime('%Y-%m-%d %H:%M')
+
+def _load_generator_characteristics(data_directory):
+    
+    # hack the _read_generators function in the RTS-GMLC parser
+    elements = {'generator': {},
+            'bus' : {'virtual_bus':{'area':'virtual_area', 'zone':'virtual_zone'}},
+            'area': {'virtual_area':{}},
+            'zone': {'virtual_zone':{}},
+            }
+    bus_id_to_names = defaultdict(lambda: 'virtual_bus')
+    parser._read_generators(data_directory, elements, bus_id_to_names)
```

Comment:
A little dangerous to use a private method, but I can live with it. Maybe you can get the guy who wrote that function to document it though...